service/src/test/java/bio/terra/externalcreds/dataAccess/SshKeyPairDaoTest.java

```diff
@@ -0,0 +1,138 @@
+package bio.terra.externalcreds.dataAccess;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.externalcreds.BaseTest;
+import bio.terra.externalcreds.generated.model.SshKeyPairType;
+import bio.terra.externalcreds.models.SshKeyPair;
+import java.util.UUID;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class SshKeyPairDaoTest extends BaseTest {
+
+  @Autowired SshKeyPairDAO sshKeyPairDAO;
+
+  private static final String DEFAULT_USER_ID = UUID.randomUUID().toString();
+  private static final SshKeyPairType DEFAULT_KEY_TYPE = SshKeyPairType.GITHUB;
+  private static final String DEFAULT_EXTERNAL_USER_EMAIL = "foo@monkeyseesmonkeydo.com";
+  private static final String DEFAULT_PRIVATE_KEY =
+      "-----BEGIN OPENSSH PRIVATE KEY-----\n"
+          + "abcde12345/+xXXXYZ//890=\n"
+          + "-----END OPENSSH PRIVATE KEY-----";
+  private static final String DEFAULT_PUBLIC_KEY =
+      "ssh-ed25519 AAAA12345 " + DEFAULT_EXTERNAL_USER_EMAIL;
+
+  @Nested
+  class UpsertKeyPair {
+    @Test
+    void testUpsertTwiceWithSameUserId() {
+      var externalUserEmail = "bar@monkeyseesmonkeydo.com";
+      var sshKeyOne = createDefaultSshKey();
+      var sshKeyTwo = createSshKey(DEFAULT_USER_ID, DEFAULT_KEY_TYPE, externalUserEmail);
```

Comment:
`sshKeyTwo` is created with the assumption that `DEFAULT_USER_ID` and `DEFAULT_KEY_TYPE` are also used in `sshKeyOne`. That should be built into the test:
```
var sshKeyTwo = sshKeyOne.withExternalUserEmail(externalUserEmail);
```